src/main/java/pl/scoutCamp/business/RankingService.java

```diff
@@ -0,0 +1,103 @@
+package pl.scoutCamp.business;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.stereotype.Service;
+import pl.scoutCamp.api.controller.enums.SortOrder;
+import pl.scoutCamp.api.controller.enums.SortType;
+import pl.scoutCamp.business.dao.TeamCategorizationSheetDAO;
+import pl.scoutCamp.business.dao.TeamDAO;
+import pl.scoutCamp.domain.Ranking;
+import pl.scoutCamp.domain.Team;
+import pl.scoutCamp.domain.TeamCategorizationSheet;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import static pl.scoutCamp.api.controller.enums.GivenCategory.*;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class RankingService {
+
+    private final TeamDAO teamDAO;
+    private final TeamCategorizationSheetDAO teamCategorizationSheetDAO;
+
+    public Page<Ranking> createRankingList(
+            String period,
+           SortType sort,
+            SortOrder order) {
+        List<Ranking> rankingList = new ArrayList<>();
+        for (Team team : findAllTeams()) {
+            Integer totalPoints = getTotalPoints(getTeamSheets(team, period));
+            Ranking teamSummary = Ranking.builder()
+                    .teamName(team.getName())
+                    .totalPoints(totalPoints)
+                    .category(getCategory(totalPoints))
+                    .sheets(getTeamSheets(team, period))
+                    .build();
+            rankingList.add(teamSummary);
+        }
+//
+        return new PageImpl<>(sortedRankingList(rankingList, sort, order));
+    }
+
+    private List<Ranking> sortedRankingList(List<Ranking> rankingList, SortType sort, SortOrder order) {
+        Comparator<Ranking> comparingByPoints = Comparator.comparing(Ranking::getTotalPoints);
+        Comparator<Ranking> comparingByName = Comparator.comparing(Ranking::getTeamName);
+        Comparator<Ranking> comparingByCategory = Comparator.comparing(Ranking::getCategory);
+        if (order.equals(SortOrder.ASC)) {
```

Comment:
You can DRY it more. Try to create "compare" method and pass there "order" parameter instead of writing IF/ELSE block with almost the same instructions.